aten/src/ATen/native/quantized/cpu/kernels/QuantizedOpKernels.cpp

```diff
@@ -2278,15 +2278,10 @@ void quantize_tensor_per_tensor_affine_cpu(
         TORCH_CHECK(
             rtensor.is_contiguous(), "Float tensor should be contiguous");
         const float* const rdata = rtensor.data_ptr<float>();
-        // If QEngine is set to QNNPACK, use caffe2 specialized Int8Quantize
-        // implementation on ARM
-        if (at::globalContext().qEngine() == at::QEngine::QNNPACK) {
-          quantize_tensor_arm<scalar_t>(
-              rdata, qtensor, rtensor.numel(), scale, zero_point);
-          return;
-        }
+        quantize_tensor_arm<scalar_t>(
```

Comment:
Will we run quantized ops on mobile without qnnpack though?

Reply:
Why does it matter? This op is not using anything from QNNPACK. It is simply arm specific implementation.

Reply:
It doesn't - I was just wondering if this fixed some issue or it was more of a cleanup.

Reply:
Just cleanup. The real issue was the macro AT_DISPATCH_QINT_TYPES, since it builds lambda from expression and inline executes it. So return inside the expression does not really return from that function and we endup going through the follow-up path as well.